src/serializer/ResidualFunctions.js

```diff
@@ -440,6 +446,11 @@ export class ResidualFunctions {
               let childBody = t.blockStatement([t.returnStatement(t.callExpression(callee, callArgs))]);
 
               funcNode = t.functionExpression(null, params, childBody);
+              if (functionValue.$Strict) {
```

Comment:
I do not think this is necessary. This function expression is just generating a stub call to the real function body so "use strict" is only needed in the real function body and unnecessary here. Let me know if you can find a testcase to cause this assumption to fail.

Reply:
Well, the test case I added fails unless I add strict to those stubs. Frankly, I'm sort of new to JS and I'm still trying to fully understand how strict works and why the test would fail without this code.

Reply:
`StrictSimple.js` fails? Interesting, but I can see how it can happen if the chain of function calls isn't all strict/non-strict. Can you also add a test case for the other case, where the function isn't strict? Just for completeness.

Reply:
`StrictComplex.js` fails if I remove this not `StrictSimple.js`.

Reply:
I added a non-strict test. LGTM?

Reply:
Now I see. Since it uses "this" point, so it goes into this code path so the strictness needs to be preserved from sub call function to the real function body. Thanks.